Src/FluentAssertions/Primitives/StringEqualityStrategy.cs

```diff
@@ -57,19 +58,47 @@ private string GetMismatchSegmentForStringsOfDifferentLengths(string subject, st
 
     public void ValidateAgainstMismatch(IAssertionScope assertion, string subject, string expected)
     {
-        if (!ValidateAgainstSuperfluousWhitespace(assertion, subject, expected) ||
-            !ValidateAgainstLengthDifferences(assertion, subject, expected))
+        if (!ValidateAgainstSuperfluousWhitespace(assertion, subject, expected))
         {
             return;
         }
 
-        int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparisonMode);
-
-        if (indexOfMismatch != -1)
+        if (expected.IsLongOrMultiline() || subject.IsLongOrMultiline())
         {
+            int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparisonMode);
+
+            if (indexOfMismatch == -1)
+            {
+                ValidateAgainstLengthDifferences(assertion, subject, expected);
+                return;
+            }
+
+            string lineInfo = $"at index {indexOfMismatch}";
+            var matchingString = subject[..indexOfMismatch];
+            int lineNumber = matchingString.Count(c => c == '\n');
+
+            if (lineNumber > 0)
+            {
+                var lastLineIndex = matchingString.LastIndexOf('\n');
```

Comment:
🤔 Did you mean `lastLineNumber`?

Reply:
No, it is the index of the last newline char before the `indexOfMismatch` and this allows me to calculate the `column` in the line.
I renamed it to `indexOfLastNewlineBeforeMismatch`.